spec/script/setup_concepts_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'spec_helper'
+require 'rails_helper'
+require 'setup_concepts'
+
+VCR.configure do |config|
+  config.cassette_library_dir = "fixtures/vcr_cassettes"
+  config.hook_into :webmock # or :fakeweb
+end
+
+describe 'Setting up concepts' do
+  VCR.use_cassette('fetching concepts') do
+
+    setup = Setup::Concepts.new
+    
+    it "should have an attribute concepts" do
+      expect(setup.concepts.class).to eq(Array)
+    end
+
+    response = setup.fetch_concepts()
+
+    it "should successfully make a http request" do
+      expect(response.code).to eq("200")
+    end
+
+    it "the response should be a string" do
+      expect(response.body.class).to eq(String)
+    end
+
+    it "can convert the string response to JSON" do
+      expect(setup.string_to_json(response.body).class).to eq(Hash)
+    end
+
+    it "hash of data has a list of concepts" do
+      expect(setup.string_to_json(response.body)["concepts"].class).to eq(Array)
+    end
+
+    it "has a hash that represents a concepts inside the list" do
+      expect(setup.concepts.first.class).to eq(Hash)
+    end
+
+    it "stores the data" do
+      expect(setup.concepts.empty?).to eq(false)
+    end
+
+    it "should have no concepts in the db at runtime" do
+      expect(Concept.count).to eq(0)
+    end
+
+    parent = {"id"=>1, "name"=>"Capitalization", "uid"=>"BRTGfOy7FGG4LB49eIxJQg", "parent_id"=>nil, "level"=>2}
+    child = {"id"=>2, "name"=>"Greece", "uid"=>"GPzZYKvZ2nSKmu7zC540bA", "parent_id"=>1, "level"=>0}
+    setup.concepts = [child, parent]
+    
+		it "can find its parent concept" do
+		  found_parent = setup.find_parent_concept_from_concepts(child)	
+      expect(found_parent).to eq(parent)
+		end
+
+    it "can find or create its parent in the database" do
+      expect(Concept.find_by_id(child["parent_id"])).to be_nil
+      found_parent = setup.find_or_create_parent_in_db(child)
+      expect(found_parent.name).to eq(parent["name"])
+    end
+
+    it "should have a valid parent if required" do
+      Concept.delete_all
+      setup.create_concept(child)
+      expect(Concept.where(name: "Greece").first.parent_id).to eq(Concept.where(name: "Capitalization").first.id)
+    end
+    
+
+    it "should be able add the concepts array into the db" do
+      expect(Concept.count).to eq(0)
+      setup.create_all
+      expect(Concept.count).to eq(setup.concepts.length)
+      expect(Concept.find_by_name("Greece").parent).to eq(Concept.find_by_name("Capitalization"))
```

Comment:
This test depends on Capitalization always being the parent of Greece, which is dependent on data living in another file. One way around this would be to split the functionality of Concept Setup into three modules - one that creates concepts given a hash of them (and doesn't know about api requests), one that makes an api request, and one that feeds the result of the latter into the former. This way you could test that first one using a very small hash that lives directly in the spec file. If I'm not making sense here let's perhaps discuss next time you are at the bathaus...